### PR TITLE
feat: notify users about contributions waiting for description

### LINKED ISSUE #413: add email to notify user to fill contribution description

### PR DESCRIPTION
**Pull request by gabrielcosta42**
This PR adds a feature to send emails to users after a contribution is approved to request a description for it.

Closes #413

### CODE REVIEW

File: app/admin/contributions.rb [hunk 1]
```diff
@@ -26,7 +26,10 @@
   end
   batch_action :approve, if: proc { params[:scope] != "recusado" && params[:scope] != "aprovado" } do |ids|
     batch_action_collection.find(ids).each do |contribution|
-      contribution.approve!(current_user.id) if contribution.received?
+      if contribution.received?
+        contribution.approve!(current_user.id)
+        NotificationMailer.notify_fill_contribution_description(contribution).deliver_now
```
Comment (adbatista) [inline app/admin/contributions.rb:31]: this should be sent when is received, trigger the email right after creating the contribution

File: app/admin/contributions.rb [hunk 2]
```diff
@@ -131,6 +134,7 @@
   controller do
     def approve
       resource.approve!(current_user.id)
+      NotificationMailer.notify_fill_contribution_description(resource).deliver_now
```
Comment (adbatista) [inline app/admin/contributions.rb:137]: also would be nice to do this async (deliver_later)

File: app/mailers/notification_mailer.rb [hunk 3]
```diff
@@ -58,4 +58,10 @@ def notify_newsletter_contributions(contributions)
     newsletter_email = ENV['NEWS_EMAIL']
     mail(to: newsletter_email, subject: 'Punchlock - Contribuições aprovadas')
   end
+
+  def notify_fill_contribution_description(contribution)
+    @user = contribution.user
+    @contribution = contribution
+    mail(to: @user.email, subject: 'Punchlock - Contribuições Aguardando Descrição')
```
Comment (adbatista) [inline app/mailers/notification_mailer.rb:65]: add i18n pls

File: config/locales/pt-BR/views.yml [hunk 4]
```diff
@@ -56,6 +56,18 @@ pt-BR:
...
+    notify_fill_contribution_description_html: |
+      <p>
+        Olá,
+      </p>
+      <p>
+        Você possui a seguinte contribuição aguardando descrição da PR.
+      </p>
+      <p>
+        Acessar PR&#58; 
+        <a href='https://%{host}/contributions/%{contribution.id}'>%{contribution.link}</a>
```
Comment (adbatista) [inline config/locales/pt-BR/views.yml:68]: you could pass the url as param, so we will keep using the URL helpers

File: spec/mailers/notification_mailer_spec.rb [hunk 5]
```diff
@@ -249,5 +249,22 @@
         expect(mail.body.encoded).to have_link('here', href: edit_user_password_url(reset_password_token: 'xyz'))
       end
     end
+
+    context 'when notify user to fill contribution description' do
```
Comment (adbatista) [inline spec/mailers/notification_mailer_spec.rb:253]: check if the email has the url

File: app/admin/contributions.rb (hunk 1 above)
Comment (gabrielcosta42) [reply app/admin/contributions.rb]: Done

File: app/admin/contributions.rb (hunk 2 above)
Comment (gabrielcosta42) [reply app/admin/contributions.rb]: Done

File: app/mailers/notification_mailer.rb (hunk 3 above)
Comment (gabrielcosta42) [reply app/mailers/notification_mailer.rb]: Done

File: config/locales/pt-BR/views.yml (hunk 4 above)
Comment (gabrielcosta42) [reply config/locales/pt-BR/views.yml]: Done

File: spec/mailers/notification_mailer_spec.rb (hunk 5 above)
Comment (gabrielcosta42) [reply spec/mailers/notification_mailer_spec.rb:253]: Done

File: app/jobs/alert_fill_contribution_description_job.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class AlertFillContributionDescriptionJob < ApplicationJob
+  queue_as :default
+
+  def perform(user_id)
```
Comment (adbatista) [inline app/jobs/alert_fill_contribution_description_job.rb:4]: missing specs

**Pull request merged**